scripts/s3_log_sync.py

```diff
@@ -3,11 +3,113 @@
+        self.src_paths = src_paths
+        self.src_logdir = src_logdir
+        self.bucket_name = bucket_name
+        self.s3_subdir = s3_subdir
+        self.profile_name = profile_name
+        self.delete = delete
+        self.workers = workers if workers else cpu_count()
+        self.s3_sync_cmd = ["aws", "s3", "sync"]
+        if self.profile_name:
+            self.s3_sync_cmd += ["--profile", self.profile_name]
+        self.s3_sync_cmd += ["--exclude", "*"]
 
+    def run(self):
+        for src_path in self.src_paths.items():
+            self.sync_directory(src_path)
 
-s3 = boto3.resource('s3')
+    def sync_directory(self, path_tuple):
+        # Use subprocess to call the AWS CLI command and display progress
+        # HACK
+        src_path = path_tuple[0]
+        path_include = path_tuple[1]['include']
+        s3_path = os.path.join(self.s3_subdir, os.path.relpath(src_path, self.src_logdir))
+        cmd = self.s3_sync_cmd + [src_path, f"s3://{self.bucket_name}/{s3_path}"]
+        for include in path_include:
+            cmd += ["--include", include]
+        start_time = time.monotonic()
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=1,
+                              universal_newlines=True) as proc:
+            for line in proc.stdout:
+                if "Completed" in line:
+                    try:
+                        line_components = line.split()
+                        sent = int(float(line_components[1]))
+                        sent_incre = line_components[2].split('/')[0]
+                        total_size = int(float(line_components[2].split('/')[1].replace("~", "")))
+                        total_size_incre =  line_components[3]
+                        sent = self.get_bytes(sent, sent_incre)
+                        total_size = self.get_bytes(total_size, total_size_incre)
+                        progress = sent / total_size
+                        print(f"{src_path} - "
+                              f"{self.convert_size(sent)} / {self.convert_size(total_size)} - "
+                              f"{progress:.0%} - "
+                              f"{self.get_throughput(sent, start_time)}", end="\r")
+                    except Exception as e:
+                        print(f"Error parsing aws cli output: {e}", end="\r")
+            print(f"Completed syncing {src_path}")
 
-for bucket in s3.buckets.all():
-    print(bucket.name)
+        if self.delete:
+            print(f"Deleting local files from {src_path}")
+            shutil.rmtree(src_path)
+
+    @staticmethod
+    def convert_size(size):
+        # Convert bytes to a human-readable format
+        if size == 0:
+            return "0B"
+        size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+        i = int(math.floor(math.log(size, 1024)))
+        p = math.pow(1024, i)
+        s = round(size / p, 2)
+        return f"{s}{size_name[i]}"
```

Comment:
The Python [humanize](https://pypi.org/project/humanize/) API provides convenient functions to do this for you; it even goes up to YB

Reply:
Will include in next PR.